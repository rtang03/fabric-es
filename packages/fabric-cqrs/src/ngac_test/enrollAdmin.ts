// /**
//  * @packageDocumentation
//  * @hidden
//  */
// require('dotenv').config({ path: './.env.ngac.org2.test' });
// import { enrollAdmin } from '@fabric-es/operator';
// import { Wallets } from 'fabric-network';
//
// // todo: this one should remove wallet before enrollAdmin
//
// enrollAdmin({
//   caUrl: process.env.ORG_CA_URL,
//   enrollmentID: process.env.ORG_ADMIN_ID,
//   enrollmentSecret: process.env.ORG_ADMIN_SECRET,
//   mspId: process.env.MSPID,
//   label: process.env.ORG_ADMIN_ID,
//   context: {
//     fabricNetwork: process.env.NETWORK_LOCATION,
//     connectionProfile: process.env.CONNECTION_PROFILE,
//     wallet: await Wallets.newFileSystemWallet(process.env.WALLET)
//   }
// })
//   .then(result => console.log(result))
//   .catch(error => {
//     console.error(error);
//     process.exit(-1);
//   });
