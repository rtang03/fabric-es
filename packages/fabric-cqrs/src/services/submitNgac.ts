import { Context } from '../types';
import { getContract } from './contract';

export const submitNgac: <T = any>(
  fcn: string,
  args: string[],
  { network }: Context
) => Promise<T | { error?: any; status?: string; message?: string }> = async <
  T = any
>(
  fcn,
  args,
  { network }
) =>
  await getContract(network).then(({ contract }) =>
    contract
      .createTransaction(fcn)
      .submit(...args)
      .then<T>((res: any) =>
        JSON.parse(Buffer.from(JSON.parse(res)).toString())
      )
      .catch(error => {
        console.log(`Error processing Submit transaction.`);
        console.error(error.stack);
        console.log(fcn);
        return { error };
      })
  );

export default submitNgac;
