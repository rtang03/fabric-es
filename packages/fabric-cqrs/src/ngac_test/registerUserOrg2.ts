import { Context, registerUser as register } from '@espresso/admin-tool';
import { ChannelEventHub } from 'fabric-client';
import { FileSystemWallet, Gateway, Network } from 'fabric-network';
import { getNetwork } from '../services';

const defaultContext: Context = {
  connectionProfile: process.env.CONNECTION_PROFILE,
  fabricNetwork: process.env.NETWORK_LOCATION,
  wallet: new FileSystemWallet(process.env.WALLET)
};

export const registerUserOrg2: (option: {
  enrollmentID: string;
  enrollmentSecret: string;
  context?: Context;
}) => any = async ({
  enrollmentID,
  enrollmentSecret,
  context = defaultContext
}) => register(enrollmentID, enrollmentSecret, context);

export const bootstrap: (
  identity: string
) => Promise<{
  identity: string;
  network: Network;
  gateway: Gateway;
  channelHub: ChannelEventHub;
}> = async identity => {
  try {
    await registerUserOrg2({
      enrollmentID: identity,
      enrollmentSecret: 'password'
    });
    return await getNetwork({ identity });
  } catch (err) {
    console.log(err);
    process.exit(-1);
  }
};
