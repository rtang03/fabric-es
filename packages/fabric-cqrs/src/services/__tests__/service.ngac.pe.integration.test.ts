import { Gateway, Network } from 'fabric-network';
import { bootstrap } from '../../ngac_test/registerUserOrg2';
import submitNgac from '../submitNgac';

let network: Network;
let gateway: Gateway;

const identity = `svs_org2_pe_test${Math.floor(Math.random() * 1000)}`;

beforeAll(async () => {
  const ctx = await bootstrap(identity);
  network = ctx.network;
  gateway = ctx.gateway;
});

afterAll(async () => await gateway.disconnect());

const mspId = 'Org2MSP';
const entityName = 'doc';
const entityId = 'doc_0001';
const mspAttrs = [{ type: '1', key: 'mspid', value: 'Org2MSP' }];
const pClass = 'ngac-pe-policy';
const sid = 'allowCreateDoc';
const url = `model/${mspId}/${entityName}`;
const condition = { hasList: { createDoc: 'creator_id' } };
const allowedEvents = ['DocCreated'];
const x509id =
  'x509::/C=US/ST=North Carolina/O=Hyperledger/OU=client/CN=Admin@org1.example.com::/C=US/ST=North Carolina/O=Hyperledger/OU=Fabric/CN=rca-org1';
const resourceAttr = [{ type: 'N', key: 'createDoc', value: [x509id] }];

describe('Ngac Permission Tests', () => {
  describe('prepare Ngac environment', () => {
    it('should create Org2MSP attribute', async () =>
      submitNgac('addMSPAttr', [mspId, JSON.stringify(mspAttrs)], {
        network
      }).then(attributes => console.log(attributes)));

    it('should create policy', async () =>
      submitNgac(
        'addPolicy',
        [
          pClass,
          sid,
          url,
          JSON.stringify(allowedEvents),
          JSON.stringify(condition)
        ],
        { network }
      ).then(policy => console.log(policy)));

    it('should create resource attributes for entityName', async () =>
      submitNgac(
        'addResourceAttr',
        [entityName, '', JSON.stringify(resourceAttr)],
        { network }
      ).then(attributes => console.log(attributes)));
  });
});
