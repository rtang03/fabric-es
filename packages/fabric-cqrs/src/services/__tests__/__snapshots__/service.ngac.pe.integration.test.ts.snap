// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ngac Permission Tests prepare Ngac environment should create Org2MSP attribute 1`] = `
Array [
  Object {
    "key": "mspid",
    "type": "1",
    "value": "Org2MSP",
  },
]
`;

exports[`Ngac Permission Tests prepare Ngac environment should create policy 1`] = `
Object {
  "allowedEvents": Array [
    "DocCreated",
  ],
  "attributes": Object {
    "uri": "model/Org2MSP/doc",
  },
  "condition": Object {
    "hasList": Object {
      "createDoc": "creator_id",
    },
  },
  "effect": "Allow",
  "policyClass": "ngac-pe-policy",
  "sid": "allowCreateDoc",
}
`;

exports[`Ngac Permission Tests prepare Ngac environment should create resource attributes for entityName 1`] = `
Array [
  Object {
    "key": "createDoc",
    "type": "1",
    "value": "Org2MSP::Admin@org2.example.com",
  },
]
`;
